software/sys_controller/ossc/av_controller.c

```diff
@@ -808,82 +813,58 @@ int main()
-                target_tvp = TVP_INPUT1;
-                target_format = FORMAT_YPbPr;
-                target_typemask = VIDEO_LDTV|VIDEO_SDTV|VIDEO_EDTV|VIDEO_HDTV;
+            } else if (target_input <= AV2_RGsB) {
                 target_ths = THS_INPUT_A;
                 target_pcm = PCM_INPUT3;
-                break;
-            case AV2_RGsB:
-                target_tvp = TVP_INPUT1;
-                target_format = FORMAT_RGsB;
-                target_typemask = VIDEO_LDTV|VIDEO_SDTV|VIDEO_EDTV|VIDEO_HDTV;
-                target_ths = THS_INPUT_A;
-                target_pcm = PCM_INPUT3;
-                break;
-            case AV3_RGBHV:
+            } else  { // if (target_input <= AV3_YPBPR) {
                 target_tvp = TVP_INPUT3;
-                target_format = FORMAT_RGBHV;
-                target_typemask = VIDEO_PC;
                 target_ths = THS_STANDBY;
                 target_pcm = PCM_INPUT2;
-                break;
+            }
+
+            switch (target_input) {
+            case AV1_RGBs:
             case AV3_RGBs:
-                target_tvp = TVP_INPUT3;
                 target_format = FORMAT_RGBS;
-                target_typemask = VIDEO_LDTV|VIDEO_SDTV|VIDEO_EDTV|VIDEO_HDTV;
-                target_ths = THS_STANDBY;
-                target_pcm = PCM_INPUT2;
                 break;
+            case AV1_RGsB:
+            case AV2_RGsB:
             case AV3_RGsB:
-                target_tvp = TVP_INPUT3;
                 target_format = FORMAT_RGsB;
-                target_typemask = VIDEO_LDTV|VIDEO_SDTV|VIDEO_EDTV|VIDEO_HDTV;
-                target_ths = THS_STANDBY;
-                target_pcm = PCM_INPUT2;
                 break;
+            case AV1_YPBPR:
+            case AV2_YPBPR:
             case AV3_YPBPR:
-                target_tvp = TVP_INPUT3;
                 target_format = FORMAT_YPbPr;
-                target_typemask = VIDEO_LDTV|VIDEO_SDTV|VIDEO_EDTV|VIDEO_HDTV;
-                target_ths = THS_STANDBY;
-                target_pcm = PCM_INPUT2;
+                break;
+            case AV3_RGBHV:
+                target_format = FORMAT_RGBHV;
+                target_typemask = VIDEO_PC;
                 break;
             default:
                 break;
             }
 
             printf("### SWITCH MODE TO %s ###\n", avinput_str[target_input]);
 
-            // The input changed, so load the appropriate profile
+            // The input changed, so load the appropriate profile if
+            // input->profile link is enabled
             if (profile_link && (profile_sel != input_profiles[target_input])) {
                 profile_sel = input_profiles[target_input];
                 read_userdata(profile_sel);
             }
 
+            // If profile->input link is enabled, update it to the new input to
+            // stay consistent, but don't automatically save the new setting.
+            if (tc.link_av != AV_LAST)
+                tc.link_av = target_input;
```

Comment:
I'm not sure if this is a good idea. If user uses profile X with inputs A and B, but wants automatic switch to A upon loading X, saving the profile while in B input reassigns auto-load to B.

Reply:
If a user uses profile X with A and B, he should not have profile->input link set. But it could happen, yes.
Maybe it's a good idea to set a FLAG in previous if condition e.g. avoid_prof_av_link = 1, which is unset after line 866.

Reply:
Hmm, I think that's a fair point, @marqs85. The idea behind the above code snippet was that we don't end up in a situation where we are on profile X, with a profile->input link X->A, but input B is actually active. OTOH, now that I think of it, this doesn't seem to really be an immediate problem in itself. However, not doing this adjustment can lead to a situation where we have an input->profile link A->X and a profile->input link X->B.

Reply:
I will remove the code snippet in question, as it may prove to be quite annoying to users trying to adjust their settings.